python/ray/data/datasource/csv_datasource.py

```diff
@@ -40,19 +40,27 @@ def _read_stream(
         if hasattr(parse_options, "invalid_row_handler"):
             parse_options.invalid_row_handler = parse_options.invalid_row_handler
 
-        reader = csv.open_csv(
-            f, read_options=read_options, parse_options=parse_options, **reader_args
-        )
-        schema = None
-        while True:
-            try:
-                batch = reader.read_next_batch()
-                table = pyarrow.Table.from_batches([batch], schema=schema)
-                if schema is None:
-                    schema = table.schema
-                yield table
-            except StopIteration:
-                return
+        try:
+            reader = csv.open_csv(
+                f, read_options=read_options, parse_options=parse_options, **reader_args
+            )
+            schema = None
+            while True:
+                try:
+                    batch = reader.read_next_batch()
+                    table = pa.Table.from_batches([batch], schema=schema)
+                    if schema is None:
+                        schema = table.schema
+                    yield table
+                except StopIteration:
+                    return
+        except pa.lib.ArrowInvalid as e:
+            raise pa.lib.ArrowInvalid(
+                f"{e}. Failed to read CSV file: {path}. "
+                "Please check the CSV file has correct format, or filter out non-CSV "
+                "file with 'partition_filter' field. See read_csv() documentation for "
+                "more details."
+            )
```

Comment:
Dumping the original exception at the beginning of this exception's message will get put a potentially wordy message before our higher-level message and will get rid of the original traceback, replaced with our traceback. Might be better to just chain the exceptions here since that's the use case for exception chaining.

Also I think that raising a `ValueError` rather than Arrow's `ArrowInvalid` error makes more sense here.
```suggestion
            raise ValueError(
                f"Failed to read CSV file: {path}. "
                "Please check the CSV file has correct format, or filter out non-CSV "
                "file with 'partition_filter' field. See read_csv() documentation for "
                "more details."
            ) from e
```

Reply:
@clarkzinzow - didn't know our best practice here. This makes sense to me. Updated.